pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -177,7 +197,15 @@ assert allPkgconfigDepends != [] -> pkgconfig != null;
 stdenv.mkDerivation ({
   name = "${pname}-${version}";
 
-  outputs = if (args ? outputs) then args.outputs else ([ "out" ] ++ (optional enableSeparateDataOutput "data") ++ (optional enableSeparateDocOutput "doc"));
+  outputs = if (args ? outputs) then args.outputs else
+    ( [ "out" ]
+    ++ (optional enableSeparateDataOutput "data")
+    ++ (optional enableSeparateDocOutput "doc")
+    ++ (optional enableSeparateBinOutput "bin")
+    ++ (optional enableSeparateBinOutput "libexec")
```

Comment:
Main reason for splitting out `libexec` was to get it out of `out` such that any references to the binaries in `libexec` from other outputs would not create `someOutput -> libexec (out) -> someOutput (propagated inputs list)`.

On the other side, I don't see a real downside to not splitting out what we can.